File: rust/src/models/address.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::country_code::CountryCode;

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct Address {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub city: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub country: Option<CountryCode>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub line1: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub line2: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub state: Option<String>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub zip_code: Option<String>,
}

impl Address {
    pub fn new() -> Self {
        Self {
            city: None,
            country: None,
            line1: None,
            line2: None,
            state: None,
            zip_code: None,
        }
    }
}

File: typescript/src/models/address.ts
// this file is @generated
import { type CountryCode, CountryCodeSerializer } from "./countryCode";

export interface Address {
  city?: string | null;

  country?: CountryCode | null;

  line1?: string | null;

  line2?: string | null;

  state?: string | null;

  zipCode?: string | null;
}

export const AddressSerializer = {
  _fromJsonObject(object: any): Address {
    return {
      city: object["city"],
      country:
        object["country"] != null
          ? CountryCodeSerializer._fromJsonObject(object["country"])
          : undefined,
      line1: object["line1"],
      line2: object["line2"],
      state: object["state"],
      zipCode: object["zip_code"],
    };
  },

  _toJsonObject(self: Address): any {
    return {
      city: self.city,
      country:
        self.country != null
          ? CountryCodeSerializer._toJsonObject(self.country)
          : undefined,
      line1: self.line1,
      line2: self.line2,
      state: self.state,
      zip_code: self.zipCode,
    };
  },
};
